src/Middleware/MigrationsMiddleware.php

```diff
@@ -0,0 +1,79 @@
+<?php
+declare(strict_types=1);
+
+namespace Migrations\Middleware;
+
+use Cake\Console\ConsoleIo;
+use Cake\Core\Configure;
+use Cake\Core\Exception\CakeException;
+use Cake\Core\InstanceConfigTrait;
+use Migrations\Config\Config;
+use Migrations\Migration\Manager;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class MigrationsMiddleware implements MiddlewareInterface
+{
+    use InstanceConfigTrait;
+
+    protected array $_defaultConfig = [
+        'paths' => [
+            'migrations' => ROOT . DS . 'config' . DS . 'Migrations' . DS,
+        ],
+        'environment' => [
+            'adapter' => null,
+            'connection' => 'default',
+            'database' => null,
+            'migration_table' => 'phinxlog',
+        ],
```

Comment:
Should the default config change when the current request is in a plugin? We could have migrations look at the plugins when a request is for a plugin.

Reply:
I think so
Given my idea on the top, I didnt want to refactor too early.
What if we have apps like mine that usually run "composer migrate" with
```
	"scripts": {
		"migrate": [
			"bin/cake migrations migrate -p Feedback --no-lock",
			"bin/cake migrations migrate -p DatabaseLog --no-lock",
			"bin/cake migrations migrate -p Queue --no-lock",
			"bin/cake migrations migrate -p QueueScheduler --no-lock",
			"bin/cake migrations migrate -p Tags --no-lock",
			"bin/cake migrations migrate -p Geo --no-lock",
			"bin/cake migrations migrate -p Captcha --no-lock",
			"bin/cake migrations migrate --no-lock" // actual app
		],
```
After all, we probably would want to be able to check on all migrations, including plugins, right?

Can we here inject a list of plugins to do on top of the app one?
My idea would be to always do app unless set to false, and plugins as list.
And we freshly apply the environment into it based on those values maybe.

Reply:
Providing a list of plugins to check is a good solution imo. I would like to eventually get away from the multiple phinxlog tables. That will have to wait until the phinx removal is complete though.